internal/venice-common/src/main/java/com/linkedin/venice/kafka/validation/Segment.java

```diff
@@ -76,6 +76,13 @@ public class Segment {
   // record the last producer message time stamp passed within the ConsumerRecord
   private long lastRecordProducerTimestamp = -1;
 
+  /**
+   * Indicate if a segment is constructed from a check point. When a host regains its leadership and consume RT,
+   * DIV tolerates the segment gap for very first message it meets and creates a new unregistered segment to replace it.
+   * Only set to true when a segment is created from a check point or copy constructed into the leader DIV.
+   */
+  private final boolean isCreatedFromCheckPoint;
```

Comment:
I understand the intention is to cover an edge case for RT DIV info; what would be the impact for VT DIV info?

Will it introduce new edge case for VT? Or VT has the same problem and this fix covers it?

Reply:
Good point and I think we can restrict the condition further to only apply to RT data, e.g.

`if (endOfPushReceived && previousSegment.isCreatedFromCheckPoint())`
-->
`
      if (endOfPushReceived && previousSegment.isCreatedFromCheckPoint() && Version.isRealTimeTopic(consumerRecord.getTopicName()))
`
what do you think?